docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,546 @@
+---
+layout: docs
+---
+
+# Introduction
+
+Oftentimes, our applications read and manipulate data from a variety of
+different sources such as databases, web services or file systems. These data
+sources are subject to latency, and we'd prefer to query them efficiently.
+
+If we are just reading data, we can make a series of optimizations such as:
+
+ - batching requests to the same data source
+ - requesting independent data from different sources in parallel
+ - caching previously seen results
+
+However, if we mix these optimizations with the code that fetches the data
+we may end up trading clarity for performance. Furthermore, we are
+mixing low-level (optimization) and high-level (business logic with the data
+we read) concerns.
+
+Fetch is a library that allows your data fetches to be written in a concise,
+composable way while executing efficiently. You don't need to use any explicit
+concurrency construct but existing idioms: applicative for concurrency and
+monad for sequencing.
+
+# Installation
+
+First of all, add the following dependency to your SBT build file:
+
+```scala
+"org.fortysevendeg" %%% "fetch" % "0.1.0"
+```
+
+Now you'll have Fetch available both in Scala and Scala.js.
+
+# Usage
+
+In order to tell Fetch how to retrieve data, we must implement the `DataSource` typeclass.
+
+```scala
+trait DataSource[Identity, Result]{
+  def fetch(ids: List[Identity]): Eval[Map[Identity, Result]]
+}
+```
+
+It takes two type parameters:
+
+ - `Identity`: the identity we want to fetch (a `UserId` if we were fetching users)
+ - `Result`: the type of the data we retrieve (a `User` if we were fetching users)
+
+The `fetch` method takes a non empty list of identities and must return an `Eval` that will result
```

Comment:
Unfortunately cats' `Eval` doesn't have docs yet, which is a shame. I'm going to take some time to contribute documentation for `Eval` and will add links when it's merged.